src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -956,7 +957,12 @@ internal static RefKind GetArgumentRefKind(ImmutableArray<BoundExpression> argum
                     argRefKind = argRefKindsOpt[i];
 
                     Debug.Assert(argRefKind == parameters[i].RefKind ||
-                            argRefKind == RefKindExtensions.StrictIn && parameters[i].RefKind == RefKind.In,
+                            parameters[i].RefKind switch
+                            {
+                                RefKind.In => argRefKind == RefKindExtensions.StrictIn,
```

Comment:
>argRefKind [](http://example.com/codeflow?start=46&length=10)

Couldn't ```argRefKind``` be ```RefKind.None``` here? Since it looks like we expect this value for the ```RefKind.RefReadOnlyParameter``` case.